docs/rfcs/opinionated_sampling_cr.md

```diff
@@ -0,0 +1,140 @@
+
+### CASE 1
+
+As a cluster administrator I want to reduce the amount of traffic caused by generating telemetry data.
+
+### CASE 2
+
+As a cluster administrator I want to be in control of the collector resources while allowing a user to change sampling policies.
+
+### CASE 3
+
+As a user I want to be able to filter relevant data without much specific open telemetry knowledge. 
+
+## Struct Design
+
+This proposal introduces a new CR which serve as an abstraction setting up an opentelemetry collector based sampling layer. Similar to the following figure.
+
+![sampling arch](./images/arch_sampling_crd.png)
+
+This custom resource creates an environment that allows us to apply e.g. tailbased sampling in a distributed environment. The operator takes care of creating an optional otel LB service and sampling instances similar to the figure shown above.
+
+LB instances will be pre-configured to distribute traces based on a given routing key like the traceID to the sampler instances.
+
+A policy is used to define which telemetry data should be sampled. Available policies can be found in the [tailbased sampling description](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/tailsamplingprocessor).
+
+```yaml
+---
+apiVersion: opentelemetry.io/v1alpha1
+kind: Sampler
+metadata:
+  name: example-sampler
+spec:
+  # Policies taken into account when making a sampling decision.
+  policies:
+    - name: "retain-error-policy"
+      type: status_code,
+      status_codes: [ERROR, UNSET]
+  # RoutingKey describes how traffic to be sampled is distributed. It can be 'traceid' or 'service'.
+  # Default is 'traceid'.
+  routingKey: traceid
+  # DecisionWait defines the time to wait before making a sampling decision.
+  # Default is 30s, specified in nanoseconds (30s = 30000000000ns).
+  decision_wait: 30000000000
+  # NumTraces defines the number of traces kept in memory.
+  # Default is 5000.
+  num_traces: 5000
+  # ExpectedNewTracesPerSec defines the expected number of new traces per second.
+  # Helps allocate memory structures. Default is 5000.
+  expected_new_traces_per_sec: 5000
+  # DecisionCache defines the settings for the decision cache.
+  # This allows sampling decisions to be cached to avoid re-processing traces.
+  decision_cache:
+    # SampledCacheSize configures the amount of trace IDs to be kept in an LRU
+    # cache, persisting the "keep" decisions for traces that may have already
+    # been released from memory.
+    # By default, the size is 0 and the cache is inactive.
+    # If using, configure this as much higher than num_traces so decisions for
+    # trace IDs are kept longer than the span data for the trace.
+    sampled_cache_size: 10000
+  # Components defines the template of all requirements to configure scheduling
+  # of all components to be deployed.
+  components:
+    loadbalancer:
+      # Defines if the component is managed by the operator
+      managementState: managed
```

Comment:
Why is it needed to set this field at the component level instead of just doing it at the full object level?

Reply:
Actually Ive no strong opinion on that.